src/AppCommon/Infra/Out.cs

```diff
@@ -32,9 +32,16 @@ public static void Write(string text)
 
     public static string ReadLine()
     {
-        var result = Console.ReadLine();
-        output.AppendLine(result);
-        return result;
+        try
+        {
+            var result = Console.ReadLine();
+            output.AppendLine(result);
+            return result;
+        }
+        catch (UnauthorizedAccessException)
```

Comment:
When would this throw? I don't understand, but I also haven't tried it.

Reply:
In an environment with a non-interactive shell, so stdin isn't available and so Console.ReadLine throws.

Reply:
That makes sense! Thanks!